src/command_modules/azure-cli-network/azure/cli/command_modules/network/generated.py

```diff
@@ -370,3 +379,28 @@
 cli_command('network traffic-manager endpoint list', list_traffic_manager_endpoints)
 register_generic_update('network traffic-manager endpoint update', EndpointsOperations.get, EndpointsOperations.create_or_update, factory)
 
+# DNS ZonesOperations
+factory = lambda _: get_mgmt_service_client(DnsManagementClient).zones
+cli_command('network dns zone show', ZonesOperations.get, factory)
+cli_command('network dns zone delete', ZonesOperations.delete, factory)
+cli_command('network dns zone list', list_dns_zones)
+register_generic_update('network dns zone update', ZonesOperations.get, ZonesOperations.create_or_update, factory)
+
+factory = lambda _: get_mgmt_service_client(DnsZoneClient).dns_zone
+cli_command('network dns zone create', DnsZoneOperations.create_or_update, factory, transform=DeploymentOutputLongRunningOperation('Starting network dns zone create'))
+
+# DNS RecordSetsOperations
+factory = lambda _: get_mgmt_service_client(DnsManagementClient).record_sets
+cli_command('network dns record-set show', RecordSetsOperations.get, factory)
+cli_command('network dns record-set delete', RecordSetsOperations.delete, factory)
+cli_command('network dns record-set create', create_dns_record_set)
+register_generic_update('network dns record-set update', RecordSetsOperations.get, RecordSetsOperations.create_or_update, factory)
+cli_command('network dns record-set record add-aaaa', add_dns_aaaa_record)
+cli_command('network dns record-set record add-a', add_dns_a_record)
+cli_command('network dns record-set record add-cname', add_dns_cname_record)
+cli_command('network dns record-set record add-ns', add_dns_ns_record)
+cli_command('network dns record-set record add-mx', add_dns_mx_record)
+cli_command('network dns record-set record add-ptr', add_dns_ptr_record)
+cli_command('network dns record-set record add-soa', add_dns_soa_record)
+cli_command('network dns record-set record add-srv', add_dns_srv_record)
+cli_command('network dns record-set record add-txt', add_dns_txt_record)
```

Comment:
Consider changing the naming convention here: 'network dns record-set cname-record add'
Putting the record type first will enable quicker tab completion than if they are all appended with add-. Also, when you add the deletes, you won't get a very dense command set on "record" (which won't need to exist at all). If only 5 people have used these commands in Xplat, there's probably no harm in deviating from whatever scheme they used (which we already have anyways).